### DISCUSSION TITLE
Nightscout on Google Cloud failing

### DISCUSSION THREAD

**Discussion by sandrine-boh** [NightscoutFoundation/xDrip#2493] [Q&A]
I am having issues with phase 2 of creating NightScout through Google Cloud. I followed the guide but when I completing phase 2 of the installation of Linux/NS, an error message appears after I have entered my email. See error message screenshot below:

<img width="948" alt="Screenshot 2022-11-23 at 1 51 49 PM" src="https://user-images.githubusercontent.com/118936789/203641317-564e0444-7322-40ba-be5d-fcb8dfeff02a.png">

**Comment by Navid200** [comment]
@tzachi-dar Would you please have a look at this?

**Comment by Navid200** [comment]
How many subdomains do you have in your FreeDNS account?
What happens if you enter the subdomain in the address bar of a web browser?

**Reply by sandrine-boh** [reply]
It asks for API_SECRET. When I enter it says that it's incorrect password. However, I copied directly from the terminal.

**Reply by sandrine-boh** [reply]
Also I only have 1 subdomain

**Reply by Navid200** [reply]
Can you, to be 100% sure, edit the nsconfig file and change the password to something much simpler with no special characters?
This is only temporary.
Then, restart the server.

After it comes back up, does it still not accept the password?
The password, like all the other variables, is inside quotation marks in the nsconfig file.

**Reply by Navid200** [reply]
Wait, I thought you created a second subdomain.
In that case, please create a second subdomain and run phase 2 with it.

**Reply by sandrine-boh** [reply]
I did create a second but proceeded to delete the first subdomain I had created so that I am left with just 1 in the freedns

**Reply by tzachi-dar** [reply]
If your issue is with wrong API_SECRET than I don't think it is related to the problem above.
Can you please do the following:
1) Reboot the machine so that your API_SECRET will be used.
2) If this does not help, please make sure that your api secret is only made from letters and numbers. Use the webtool to enter it.
Then reboot the machine, and try connecting again.

**Reply by sandrine-boh** [reply]
OK - so I have corrected any issues with API_SECRET. Here is what my NS looks like when I put in browser

<img width="1326" alt="Screenshot 2022-11-23 at 5 04 45 PM" src="https://user-images.githubusercontent.com/118936789/203653760-a8452ef3-861c-429d-b432-ef9de977a94a.png">

**Reply by Navid200** [reply]
When you click in the address bar a second time, does it populate https at the beginning?

**Comment by marionbarker** [comment]
We just saw that error message helping the Loop and Learn mentors team do a group build. The person having trouble entered a subdomain that started with a capital letter. She went to FreeDNS and deleted that one and made a new one (same name, all lower case). She restarted with menu item 3, said to use existing certificate and it went through. 

We also found that the freedns user name must be entered all lowercase in the scripts. 

It might have been enough just to enter domain all lowercase.

**Reply by Navid200** [reply]
Thanks

I don't believe FreeDNS allows you to choose a subdomain that is not all lower case.

I don't understand how this is possible as the dialog should not accept a subdomain that is not on FreeDNS.
I need more info to understand if we need to fix a bug in the script or add something to the guide or what.

As is< I don't see how it would be possible for anyone to create a subdomain that is not all lowercase.

**Reply by sandrine-boh** [reply]
My subdomain did not contain any capital letters. In addition, freedns username was also entered all in lowercase. Neither of these were the cause of the error message I am receiving.

**Reply by tzachi-dar** [reply]
[sandrine-boh](https://github.com/sandrine-boh) are you still having the problem with password? Did you try my suggestions from above.

**Reply by sandrine-boh** [reply]
@tzachi-dar @Navid200 No more issues with password. However, I am confused whether the phase 2 installation completed despite the error message. I have sent you a screenshot of what NS looks like when I enter my subdomain in browser and now successfully enter my API_SECRET. Based on the image, can you confirm whether or not phase 2 installation was successful ?

**Reply by sandrine-boh** [reply]
@tzachi-dar <img width="1308" alt="Screenshot 2022-11-23 at 5 00 14 PM" src="https://user-images.githubusercontent.com/118936789/203654781-375a8a1a-0824-49ce-a5db-f45e0f207130.png">

**Reply by tzachi-dar** [reply]
[sandrine-boh](https://github.com/sandrine-boh) seems that your domain issues are solved (unless there is a message saying non secure key above). How are you trying to upload the data?

**Reply by sandrine-boh** [reply]
@tzachi-dar as a matter of fact, there is a non-secure key and @Navid200 the https does not populate.

**Reply by tzachi-dar** [reply]
@sandrine-boh seems that in two hours you will be able to run the tool again, hopefully with better successes. Let's try that.

**Reply by Navid200** [reply]
@sandrine-boh Any updates?

**Comment by sandrine-boh** [comment]
Still the same. Now is there a possibility that I can proceed with installation despite the error message ? What would be my next step ? According to guide, xdrip but how do I get xdrip ?

**Reply by tzachi-dar** [reply]
You can use xDrip without nightscout what so ever. See for example: https://xdrip.readthedocs.io/en/latest/
Regarding finishing your installation of nightscout.
I think that your best chances are:
1) Create a new account (probably with also a new mail) on free dns, make sure to complete everything there according to the guide and continue with the new name.
2) Delete the VM, create another one and try running everything again.

HTH
Tzachi

**Reply by Navid200** [reply]
I have had 5 subdomains at the same time in FreeDNS.  I don't understand this problem.

If I were you, I would create a new subdomain leaving the previous one alone.
Then, run step 2.  it will tell you tat you have more than one subdomain and asks you to choose the one you want to use.  Enter the new one.
It should work.

Also, would you please post a screenshot of the status page?

**Reply by Navid200** [reply]
@sandrine-boh Is your subdomain all lowercase?
I didn't know it needed to be until a few hours ago.  I have added a note to the guide explaining that on the page about FreeDNS?
By any chance, do you have capital letters in your subdomain name?

**Reply by sandrine-boh** [reply]
I do not have any capital letters. I will try what you suggested above and create a new FreeDNS. But would like to keep current Google cloud account with current email I was using. Meaning, do you think it can cause issues if FreeDNS account and Google cloud accounts email are different ?

**Reply by Navid200** [reply]
Would you please create a new subdomain?  But, please pick something that is unlikely to have been used by anyone else.

**Reply by sandrine-boh** [reply]
I started over with a new email, new freeDNS account, new gmail account, followed all the steps and now another issue has occurred. I tried with another time zone and the same problem has arisen. 

<img width="376" alt="Screenshot 2022-11-27 at 11 06 19 AM" src="https://user-images.githubusercontent.com/118936789/204152253-0440c25c-f997-42e6-b55b-1bd14c6cf85b.png">

**Reply by Navid200** [reply]
Choose Oregon.

**Reply by sandrine-boh** [reply]
OK - It seems that I have managed to complete the installation by starting over with a new email for both the google cloud and freedns. Nightscout is functional. Now, how do I proceed to remote overrides/remote bolus ?

**Reply by Navid200** [reply]
Great.  Those are loop-related, right?
The loop and learn team can help you better with those I think

**Comment by marionbarker** [comment]
There are loop suggestions summarized here
* https://docs.google.com/document/d/1maCuCxFv0Ho0d1VPJYsVmPFzrP6LuqTOk3QTPEvNwks/edit?usp=share_link

But for remote carb/bolus you need LoopDocs: read and follow the links within this section
* https://loopkit.github.io/loopdocs/loop-3/features/#remote-carb-bolus


And read this whole page
* https://loopkit.github.io/loopdocs/nightscout/remote-overrides/
